Firestore/core/src/firebase/firestore/util/async_queue.h

```diff
@@ -109,22 +131,28 @@ class AsyncQueue : public std::enable_shared_from_this<AsyncQueue> {
   // calling `Enqueue` is a no-op.
   void Enqueue(const Operation& operation);
 
-  // Like `Enqueue`, but also starts the shutdown process. Once the shutdown
-  // process has started, calling any Enqueue* methods becomes a no-op
+  // Puts the `AsyncQueue` into restricted mode, where calling any Enqueue*
+  // methods becomes a no-op.
+  //
+  // The exception is `EnqueueEvenWhileRestricted`, where operations are still
```

Comment:
Optional: perhaps rephrase to something like `which would still enqueue operations even...`? (basically, active vs. passive voice).